### PR TITLE
Wdfn 504 - Changes made to clean up hydrograph rendering and to make it possible for the graph server to determine when the hydrograph rendering is complete

### PR DESCRIPTION
**Pull request by mbucknell**
Before making a pull request
----------------------------

- [X] Make sure all tests run
- [X] Update the changelog appropriately

Title
-----------
Changes required to get the graph server to work with the refactored code.

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

File: assets/src/scripts/monitoring-location/components/hydrograph/index.js [hunk 1]
```diff
@@ -96,12 +97,17 @@ export const attachToNode = function(store,
...
+    const fetchFloodLevelsPromise = store.dispatch(floodDataActions.retrieveWaterwatchData(siteno));
 
-    fetchDataPromise.then(() => {
+    let fetchDataPromises = [fetchHydrographDataPromise];
+    // If flood levels are to be shown then wait to render the hydrograph until those have been fetched.
+    if (parameterCode === config.GAGE_HEIGHT_PARAMETER_CODE) {
+        fetchDataPromises.push(fetchFloodLevelsPromise);
+    }
+    Promise.all(fetchDataPromises).then(() => {
+        console.log('Done fetching data');
```
Comment (abriggs-usgs) [inline assets/src/scripts/monitoring-location/components/hydrograph/index.js]: Seems like this log statement might be intentional, but not sure.

**Review by abriggs-usgs — APPROVED** [review]
Looks good to me.

File: assets/src/scripts/monitoring-location/components/hydrograph/index.js [hunk 2]
```diff
@@ -55,6 +55,7 @@ export const attachToNode = function(store,
                                          showOnlyGraph = false,
                                          showMLName = false
                                      } = {}) {
+    console.log('Rendering hydrograph');
```
[the remark this answers is not in the document]
Comment (mbucknell) [reply assets/src/scripts/monitoring-location/components/hydrograph/index.js]: Removed

File: assets/src/scripts/monitoring-location/components/hydrograph/index.js (hunk 1 above)
Comment (mbucknell) [reply assets/src/scripts/monitoring-location/components/hydrograph/index.js]: Removed

**Pull request merged**